common/txmgr/confirmer.go

```diff
@@ -305,6 +310,10 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) pro
 	}
 
 	ec.lggr.Debugw("Finished EnsureConfirmedTransactionsInLongestChain", "headNum", head.BlockNumber(), "time", time.Since(mark), "id", "confirmer")
+	mark = time.Now()
+
+	ec.abandonedTracker.HandleAbandonedTransactions(ctx, ec.enabledAddresses)
```

Comment:
enabledAddresses in confirmer are only changed once, inside startInternal().
As a possible optimization, from within that function, just set the enabledAddresses on the AbandonedTracker once. Then you won't need to provide it everytime you process a new head.